test/SILOptimizer/predictable_memopt_ownership.sil

```diff
@@ -1132,3 +1132,51 @@ bb9:
   %9999 = tuple()
   return %9999 : $()
 }
+
+// load [take] invalidates load forwarding. Make sure we do not try to promote
+// the load_borrow below. We can promote the load [copy] though since we restore
+// 3a.
+//
+// CHECK-LABEL: sil [ossa] @load_take_invalidates : $@convention(thin) (@owned Builtin.NativeObject) -> @owned Optional<Builtin.NativeObject> {
+// CHECK: load_borrow
+// CHECK: } // end sil function 'load_take_invalidates'
```

Comment:
You might want to add a unit test for the copy_addr code path too.

Reply:
Done.